lib/clients/stepper_control/stepper_control.py

```diff
@@ -0,0 +1,70 @@
+from common.lib.clients.qtui.QCustomSpinBox import QCustomSpinBox
+from twisted.internet.defer import inlineCallbacks
+from PyQt4 import QtGui
+
+
+class stepper_control(QtGui.QWidget):
+
+    def __init__(self, reactor):
+        super(stepper_control, self).__init__()
+        self.setSizePolicy(QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Fixed)
+        self.angle = 0.0
+        self.connect()
+
+    @inlineCallbacks
+    def connect(self):
+        """Creates an Asynchronous connection to stepper server and
+        connects incoming signals to relavent functions
+
+        """
+
+        from labrad.wrappers import connectAsync
+        self.cxn = yield connectAsync()
+        self.server = self.cxn.stepper_motor_server
+        self.initializeGUI()
+
+    def initializeGUI(self):
+        layout = QtGui.QGridLayout()
+
+        zerowidget = QtGui.QPushButton('Zero')
+        self.degreewidget = QCustomSpinBox('Angle', (-10000, 10000.0))
+        self.angle_label = QtGui.QLabel('0.0')
+        self.degreewidget.setStepSize(1.8)
+        self.degreewidget.spinLevel.setDecimals(3)
+
+        self.degreewidget.spinLevel.setValue(self.angle)
+
+        self.degreewidget.spinLevel.valueChanged.connect(self.change_angle)
+        zerowidget.clicked.connect(self.zero_scale)
+
+        layout.addWidget(zerowidget, 0, 0)
+        layout.addWidget(self.degreewidget, 0, 1)
+        layout.addWidget(self.angle_label, 0, 2)
+
+        self.setLayout(layout)
+
+    def zero_scale(self, state):
+        self.angle = 0.0
+        self.degreewidget.spinLevel.setValue(0.0)
+
+    @inlineCallbacks
+    def change_angle(self, angle):
+        difference = angle - self.angle
+        steps = difference/1.8
```

Comment:
I agree but I think the motor server should not know anything about angles, it just steps through. In the end I will most likely write a waveplate server that keeps track of and moves wave plates but in the interest of getting something usable this is a small amount of functionality in the GUI